plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedView.java

```diff
@@ -77,6 +136,102 @@ protected String getStorageMetadataLocation(String materializedViewName)
+        assertUpdate("CREATE MATERIALIZED VIEW " + viewName + " AS SELECT * FROM TABLE(mock.system.sequence_function())");
+
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("STALE");
+        assertThat(computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isNull();
+        int result1 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+
+        int result2 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result2).isNotEqualTo(result1); // differs because PTF sequence_function is called directly as mv is considered stale
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("STALE");
+        assertThat(computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isNull();
+
+        assertUpdate("REFRESH MATERIALIZED VIEW " + viewName, 1);
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("UNKNOWN");
+        ZonedDateTime lastFreshTime = (ZonedDateTime) computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue();
+        assertThat(lastFreshTime).isNotNull();
+        int result3 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result3).isNotEqualTo(result2);  // mv is not stale anymore so all selects until next refresh returns same result
+        int result4 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        int result5 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result4).isEqualTo(result3);
+        assertThat(result4).isEqualTo(result5);
+
+        assertUpdate("REFRESH MATERIALIZED VIEW " + viewName, 1);
+        assertThat((ZonedDateTime) computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isAfter(lastFreshTime);
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("UNKNOWN");
+        int result6 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result6).isNotEqualTo(result5);
+    }
+
+    @Test
+    public void testMaterializedViewCreatedFromPtfAndTable()
+    {
+        String sourceTableName = "source_table_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + sourceTableName + " (VALUE INTEGER)");
+        assertUpdate("INSERT INTO " + sourceTableName + " VALUES(2)", 1);
```

Comment:
```suggestion
        assertUpdate("CREATE TABLE " + sourceTableName + " (value integer)");
        assertUpdate("INSERT INTO " + sourceTableName + " VALUES 2", 1);
```

Reply:
I agree with removing () around 2 but type should be uppercase imho